src/ripple/basics/Buffer.h

```diff
@@ -40,8 +40,29 @@ class Buffer
 
 public:
     Buffer() = default;
-    Buffer (Buffer const&) = delete;
-    Buffer& operator= (Buffer const&) = delete;
+
+    Buffer& operator= (Buffer const& other)
+    {
+        size_ = other.size_;
+        if (size_)
+        {
+            p_.reset (new std::uint8_t[size_]);
+            memcpy (p_.get(), other.p_.get(), size_);
+        }
+        else
+            p_.reset();
+        return *this;
+    }
+
+    Buffer (Buffer const& other)
+        : size_ (other.size_)
+    {
+        if (size_)
+        {
+            p_.reset (new std::uint8_t[size_]);
```

Comment:
This is why it's a mistake to create classes with defective APIs. You will never understand the class as well later as you did when you wrote it. And god help the person who has to fix the mess you left behind without your knowledge. This class claims to be a cleaner replacement for a vector of bytes, but it's not. We should try to learn from this mistake.